### PR TITLE
[CXF-7093] Change the spring boot example to use samples parent

### PR DESCRIPTION
**Pull request by johnament**

### CODE REVIEW

File: distribution/src/main/release/samples/jax_rs/spring_boot/README [hunk 1]
```diff
@@ -8,7 +8,7 @@ The sample uses Maven. It can be built and run from the command line:
 ----
 $ mvn -Pserver
 or
-$ java -jar target/spring-boot-sample-rs-cxf-3.2.0-SNAPSHOT.jar
+$ java -jar target/spring-boot-sample-rs-cxf-*.jar
```
Comment (johnament) [inline distribution/src/main/release/samples/jax_rs/spring_boot/README]: I figure this makes it more generic and version in-specific

File: distribution/src/main/release/samples/jax_rs/spring_boot/README [hunk 2]
```diff
@@ -31,10 +31,3 @@ or access it from the CXF Services page:
 
   http://localhost:8080/services/helloservice/services
   and follow a Swagger link.
-
-To run the client from a command line open a new terminal window and run:
```
Comment (johnament) [inline distribution/src/main/release/samples/jax_rs/spring_boot/README]: This wasn't working for me even before the change.

File: distribution/src/main/release/samples/pom.xml [hunk 3]
```diff
@@ -99,6 +99,7 @@
         <module>wsdl_first_rpclit</module>
         <module>jms_pubsub</module>
         <module>jax_rs/spring_security</module>
+        <module>jax_rs/spring_boot</module>
```
Comment (johnament) [inline distribution/src/main/release/samples/pom.xml]: I only added this one for now.  If you're in agreement, I'll also apply the changes to the auto configure example.

File: distribution/src/main/release/samples/jax_rs/spring_boot/README (hunk 1 above)
Comment (reta) [inline distribution/src/main/release/samples/jax_rs/spring_boot/README]: If I am not mistaken, this command may not work, you have to provide the exact runnable: `java -jar `ls target/*.jar`` (but than it won't work for Windows users)

File: distribution/src/main/release/samples/jax_rs/spring_boot/pom.xml [hunk 4]
```diff
@@ -2,24 +2,31 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <parent>
-        <groupId>org.springframework.boot</groupId>
-        <artifactId>spring-boot-starter-parent</artifactId>
-        <version>1.3.6.RELEASE</version>
-    </parent> 
+        <groupId>org.apache.cxf.samples</groupId>
```
Comment (reta) [inline distribution/src/main/release/samples/jax_rs/spring_boot/pom.xml]: This is good one, I had it on my least as well, will probably change other samples :-), thanks.

File: distribution/src/main/release/samples/pom.xml (hunk 3 above)
Comment (reta) [inline distribution/src/main/release/samples/pom.xml]: As for me, makes sense to update other samples, @sberyozkin what do you think? Thanks, @johnament !

File: distribution/src/main/release/samples/jax_rs/spring_boot/README (hunk 1 above)
Comment (johnament) [inline distribution/src/main/release/samples/jax_rs/spring_boot/README]: @reta Ok, makes sense.  I'll change the build output to not include version #.

File: distribution/src/main/release/samples/jax_rs/spring_boot/src/main/java/sample/rs/client/SampleRestClientApplication.java [hunk 5]
```diff
@@ -31,10 +31,7 @@
 @SpringBootApplication
 @EnableJaxRsWebClient
 @EnableJaxRsProxyClient
-public final class SampleRestClientApplication {
-    private SampleRestClientApplication() {
-    }
-    
+public class SampleRestClientApplication {
```
Comment (johnament) [inline distribution/src/main/release/samples/jax_rs/spring_boot/src/main/java/sample/rs/client/SampleRestClientApplication.java]: So... this sample wasn't actually working.  The client was not defined properly.  This makes it work now

File: distribution/src/main/release/samples/jax_rs/spring_boot_scan/application/README [hunk 6]
```diff
@@ -19,9 +19,11 @@ $ mvn spring-boot:run
 
 2. Run the server 
 ----
-$ mvn -Pserver
+$ mvn spring-boot:run
```
Comment (johnament) [inline distribution/src/main/release/samples/jax_rs/spring_boot_scan/application/README]: Wherever possible, I got rid of the unnecessary profiles.

File: distribution/src/main/release/samples/jax_rs/spring_boot_scan/client/pom.xml [hunk 7]
```diff
@@ -43,22 +52,23 @@
             <artifactId>org.qi4j.library.circuitbreaker</artifactId>
             <version>${cxf.zest.version}</version>
         </dependency>
+        <dependency>
```
Comment (johnament) [inline distribution/src/main/release/samples/jax_rs/spring_boot_scan/client/pom.xml]: Ok, this dependency was a bit of an annoyance.  Its defined in the samples parent, but it interferes with spring boot's use of logback.

**Comment by johnament** [comment]
Sorry - just realized the commits weren't squashed.  Fixed

**Comment by reta** [comment]
IMO, looks cleaner, thanks @johnament, leaving decision to @sberyozkin as he authored or/and worked on these examples.

**Comment by sberyozkin** [comment]
John, does the demo work without a version qualifier of the final jar, when it is started with a "java -jar" ? Also as a side note I'm not convinced -Pserver does not make sense for the server only applications as it hides the specific plugin options - but I'm fine with getting rid of them whenever possible

**Comment by johnament** [comment]
Well we can make spring boot the default goal.  I'm not sure about your question though.  This just changes the file name and I've been able to run the examples with these instructions....

**Comment by sberyozkin** [comment]
While I appreciate this quality patch I'd like to say that the way the spring_boot_scan server and client applications are started is really off topic for CXF-7093, however as I noted I can def live with asking users to do 'mvn spring-boot:run' again instead of -Pserver or -Pclient. Re "java -jar" - I'll experiment tomorrow....

**Comment by johnament** [comment]
I'm not sure what you mean.  That didn't change other than the way mvn is called - mvn -Pclient/server vs mvn spring-boot:run for all 3 modules.  If you want me to make that work I can.  The problem is that the wrong spring boot version was getting used..

**Pull request closed without merging**